File: c_ds/train/basics/fibonacii.py

class Solution:
    def fib(self, n: int) -> int:
        if n <= 0:
            return 0
        fibos = [0 for _ in range(n+1)]
        fibos[1] = 1
        for index in range(2, n+1):
            fibos[index] = fibos[index - 1] + fibos[index - 2]
        return fibos[n]


def main():
    s = Solution()
    print(s.fib(5))
    print(s.fib(-1))
    print(s.fib(1))
    print(s.fib(100))


if __name__ == "__main__":
    main()

File: c_ds/train/basics/fibonacii.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    long long fib(int n) {
        if (n <= 0) return 0;
        long long prev = 1;
        long long prevm1 = 0;
        long long curr = 1;
        for (int pos = 2; pos <= n; pos++){
            curr = prev + prevm1;
            prevm1 = prev;
            prev = curr;
        }
        return curr;
    }
};


int main(){
    Solution s;
    cout << s.fib(5) << endl;
    cout << s.fib(-1) << endl;
    cout << s.fib(1) << endl;
    cout << s.fib(1000) << endl;
    return 0;
}